esp-wifi/src/lib.rs

```diff
@@ -290,7 +314,7 @@ impl EspWifiTimerSource for TimeBase {
 /// # }
 /// ```
 pub fn initialize(
-    init_for: EspWifiInitFor,
+    init_for: EspWifiFor,
     timer: impl EspWifiTimerSource,
     rng: hal::rng::Rng,
     radio_clocks: hal::peripherals::RADIO_CLK,
```

Comment:
I'm guessing we'll need to pass peripherals as `impl Peripheral` and bind their lifetimes in `EspWifiInitialization` so that they aren't reused before deinited. Then we'd pass `EspWifiInitialization` to the deinit functions, which may return a different `EspWifiInitialization` if they only partially deinit, or `()` if everything was dropped.

Reply:
I think we'll tackle this in a follow up PR.